### PR TITLE
docs: Clarify the use of FlowControlSettings in SusbcriberClient.

### PR DESCRIPTION
**Pull request by jskeet**
Second part of fixing the documentation aspect of #10358.

### CODE REVIEW

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs [hunk 1]
```diff
@@ -27,8 +27,15 @@ public sealed class Settings
...
         /// <summary>
         /// Flow control settings.
-        /// If <c>null</c>, uses flow control settings from <see cref="DefaultFlowControlSettings"/>. 
+        /// If <c>null</c>, uses flow control settings from <see cref="DefaultFlowControlSettings"/>.
         /// </summary>
+        /// <remarks>
+        /// Note that flow control is applied on a per-<see cref="SubscriberServiceApiClient"/> basis: a single
+        /// <see cref="SubscriberClient"/> creates multiple instances of <see cref="SubscriberServiceApiClient"/>,
+        /// and each will observe the flow control settings independently. This means there may be more
+        /// outstanding messages (or bytes) than expected; there is currently on way of limiting the
```
Comment (Rishabh-V) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs:36]: on way => no way.

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs [hunk 2]
```diff
@@ -27,8 +27,15 @@ public sealed class Settings
     {
         /// <summary>
         /// Flow control settings.
-        /// If <c>null</c>, uses flow control settings from <see cref="DefaultFlowControlSettings"/>. 
+        /// If <c>null</c>, uses flow control settings from <see cref="DefaultFlowControlSettings"/>.
         /// </summary>
+        /// <remarks>
+        /// Note that flow control is applied on a per-<see cref="SubscriberServiceApiClient"/> basis: a single
```
Comment (Rishabh-V) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs:33]: `:` => `;`?

**Review by Rishabh-V — APPROVED** [review]

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs (hunk 2 above)
Comment (amanda-tarafa) [inline apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs:33]: > `/// Note that flow control is applied on a per-<see cref="SubscriberServiceApiClient"/> basis`

Well it is and it isn't, which is somewhat confusing.

Flow control is applied on a per SubscriberServiceApiClient basis for limiting the pulled messages. But it is applied on a per SubscriberClient basis for message processing. This means that there not only may be more outstanding messages than expected (up to clientCount*maxOutstandingMessages) but also that we are not processing concurrently more than maxOutstandingMessages. I think this disparity is what's most confusing (including for #10358).

Comment (jskeet) [reply apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs:33]: Righto - will chat with you more tomorrow and try to expand this.

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs (hunk 1 above)
Comment (AAMARAB) [reply apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs]: ok

File: apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs (hunk 2 above)
Comment (jskeet) [reply apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.Settings.cs]: Okay, I've had another go at this - PTAL.

**Review by amanda-tarafa — APPROVED** [review]
Yep, this looks good.

**Review by Rishabh-V — APPROVED** [review]
Thanks Jon. This looks good. I had missed the detail in my first pass.

**Pull request merged**